go/service/user_handler.go

```diff
@@ -77,6 +88,30 @@ func (r *userHandler) passphraseStateUpdate(m libkb.MetaContext, cli gregor1.Inc
 	return nil
 }
 
+func (r *userHandler) userBlocked(m libkb.MetaContext, cli gregor1.IncomingInterface, category string, item gregor.Item) error {
+	m.Debug("userHandler: %s received", category)
+	var msg keybase1.UserBlockedGregorBody
+	if err := json.Unmarshal(item.Body().Bytes(), &msg); err != nil {
+		m.Warning("error unmarshaling user.blocked item: %s", err)
+		return err
+	}
+	m.Debug("Got user.blocked item: %+v", msg)
+	badUIDs := make(map[keybase1.UID]bool)
+	for _, r := range msg.Blocks {
+		if (r.Chat != nil && *r.Chat) || (r.Follow != nil && *r.Follow) {
+			badUIDs[r.Uid] = true
```

Comment:
Just want to make sure we've thought about the implications of treating users as a "bad user" that have `r.Follow` but not `r.Chat` blocks set. It seems that in the context of this PR it's all working properly (the user doesn't appear in recommendations). But the future should be considered.

Reply:
I just figured if you clicked block on the chat, you didn't want to then see them on your home screen.